core/primitives/src/serialize.rs

```diff
@@ -56,6 +56,13 @@ pub fn from_base_buf(s: &str, buffer: &mut Vec<u8>) -> Result<(), Box<dyn std::e
     }
 }
 
+pub fn to_str_or_base(v: &[u8]) -> String {
```

Comment:
I don’t see where this function is used. Do we need it? In fact, I feel that this function should never be used since there is no way to invert it (i.e. once you convert an of bytes to a string this way, you won’t be able to distinguish whether it was a base string originally or non-Unicode string which we converted to base)

Reply:
it was suppose to be for error output, but yeah removing